### ISSUE TITLE
Level 3 Optional methods not found 

### ISSUE THREAD

**Issue by muhdjabir** [nus-cs2030/2122-s2#186]
Hello, I'm having issues with the code crunch. It says that certain optional methods are not found whilst in works on my jshell. Does anyone have any clue as to what's going on ?

Screenshots (if any)
![Screenshot 2022-03-31 223803](https://user-images.githubusercontent.com/99940885/161081639-07a1522e-965f-458b-9f2e-4e1cb6bfdf98.png)

**Comment by yizziepeasy** [comment]
Yes, as mentioned by @chloeoxe, Prof Henry has mentioned not to use isPresent() in lecture iirc. This could be due to readability of code (lengthy checking process) and possibly better ways to use Optional class to do things.

In my opinion, Optional's best property is not hiding nulls or telling that our method can return something or not. The biggest feature that Optional does is transforming. We do not simply check if an Optional object is empty or not. We can transform it to what we want (as many times as we want) in a few lines and provide alternative solutions when one of the steps of transforming is not returning existing results. 

So, instead of checking whether holder exists with isPresent(), you can try getting holder and then use the orElse method to do the alternative. For example, if you are trying to get a module from the ImmutableMap, but it doesn't exist so yo use the orElse method to create the module. :)

**Comment by Bearypop** [comment]
Hi, adding on, the prof has said before that he banned the 3 methods: isPresent(), isEmpty(), and get() because he wants the current labs to be done using functional programming and not imperative style.

**Comment by celine-chung** [comment]
Just to add on, the methods isPresent(), IsEmpty(), and get() are imperative because they give you information about the state of the program (like whether or not the optional is empty). Instead of being state-mindful, I think the methods you are looking for would be ifPresentOrElse() and OrElse() where you explicitly tell the program what instructions to execute! Hope this helps furthers your understanding based on what everyone else has shared!
